ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/CommonPartitionEvaluator.java

```diff
@@ -263,14 +313,29 @@ public OptimizingType getOptimizingType() {
     return optimizingType;
   }
 
+  /** Segment files has enough content */
+  public boolean enoughContent() {
+    return segmentFileSize > config.getTargetSize();
+  }
+
+  /**
+   * There is at least one merge task
+   *
+   * <p>Compare the total size of the two smallest segment files and the target size
+   */
+  public boolean hasMergeTask() {
+    return min1SegmentFileSize + min2SegmentFileSize < config.getTargetSize();
+  }
+
   public boolean isMajorNecessary() {
-    return rewriteSegmentFileSize > 0;
+    return enoughContent() || rewriteSegmentFileCount > 0;
   }
 
   public boolean isMinorNecessary() {
     int smallFileCount = fragmentFileCount + equalityDeleteFileCount;
     return smallFileCount >= config.getMinorLeastFileCount()
-        || (smallFileCount > 1 && reachMinorInterval());
+        || (smallFileCount > 1 && reachMinorInterval())
+        || rewritePosSegmentFileCount > 0;
```

Comment:
This seems to be a bug fix, and we forgot to check the `rewritePosSegmentFileCount`.

Reply:
Yes, this is a bugfix.